### DISCUSSION TITLE
importScripts

### DISCUSSION THREAD

**Discussion by unicomp21** [esfx/esfx#43] [Q&A] [answered]
@rbuckton is there an example somewhere I can mimick for using esfx inside web workers?  Would be really nice too, if there was a way to handle everything where importScripts only needs to be called once w/ a single url.

**Comment by rbuckton** [comment, marked as answer]
I haven't yet determined how this would work outside of using a bundler like webpack. One option I've been considering (starting with c5591807ff4138512f7a804a50b39d33423146dd) is supporting both CommonJS and ES module emit, so that a web worker could leverage dynamic `import()` (or regular `import` declarations via `new Worker(url, { type: "module" })`). This isn't ready yet, but hopefully I'll have some time to sort this out in the next few months after I've wrapped up an actual 1.0 release.

**Comment by unicomp21** [comment]
@rbuckton should I put something on the backlog for this one?

**Reply by rbuckton** [reply]
As of the latest prerelease (`1.0.0-pre.42`), all of the esfx packages should support ES Module imports via NodeJS's subpath `exports` maps, so it may be feasible to load them via dynamic import on the web. This scenario isn't something I've had the time to test fully yet, however.
